src/watch.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{cmd, util::Fallible};
+
+use clap::ArgMatches;
+use notify::Watcher;
+use std::{
+    collections::HashSet, env, path::PathBuf, sync::mpsc::channel, thread::sleep, time::Duration,
+};
+
+pub fn watch(
+    args: ArgMatches,
+    watch_sources: HashSet<PathBuf>,
+    watch_code: HashSet<PathBuf>,
+) -> Fallible {
+    trigger_on_change(&watch_sources, &watch_code, |paths| {
+        let is_forward = paths
+            .first()
+            .map(|p| watch_sources.contains(p))
```

Comment:
Not really clear to me how to distinguish the two watchers. Is there any better way to detect where the event came from?

Reply:
Use two distinct channels for both watchers. As selection is not possible using `std`'s channels anymore, spawn separate threads to merge results using an `enum`, e.g.
```rust
enum ChangeType {
  Sources,
  Code
}

spawn(move || {
  for event in rx_sources {
    tx_changes.send(ChangeType::Source).unwrap();
  }
});


spawn(move || {
  for event in rx_code {
    tx_changes.send(ChangeType::Code).unwrap();
  }
});
```

Would have been simpler if they'd used `crossbeam-channel` which does support `select!`, but then again, `notify` spawns a thread per watcher anyway.

Reply:
I do not really understand how these threads relate to the loop we currently have.

Reply:
Ok, I think I got it...

Reply:
Not sure if I got that correctly, combined with https://github.com/mlange-42/yarner/pull/162#discussion_r589231253, but at least it works.